### ISSUE TITLE
Quit Deployment if PreFlight Job Failed

### ISSUE THREAD

**Issue by rajivml** [argoproj/argo-cd#4718]
# Summary

Hi,

I have a requirement where I need to validate the inputs by running a container and if all the inputs are right then only trigger the actual deployment.

So, I have modeled my pre-flight checks as a job and am sequencing it through **helm.sh/hook-weight** . But now the problem am facing is,  kubernetes doesn't take into account the exit status of the script i.e. triggered via kubernetes job and so does ArgoCD as a result even if pre-flight checks are failing the deployment is getting triggered which I want to avoid

![image](https://user-images.githubusercontent.com/46923567/97744186-d1e5fc00-1b0c-11eb-831b-ffb9df958400.png)

Any Ideas on how can I achieve this in argoCD ?

**Comment by jessesuen** [comment]
> But now the problem am facing is, kubernetes doesn't take into account the exit status of the script i.e. triggered via kubernetes job

Yes it does.

> so does ArgoCD as a result even if pre-flight checks are failing the deployment is getting triggered

If a job fails, Argo CD will refuse to proceed to the next phase or sync wave. I think the problem is your job is probalby exiting with code 0, otherwise your job would show up failed.

**Comment by jessesuen** [comment]
> I think the problem is your job is probalby exiting with code 0, otherwise your job would show up failed. 

Please paste job object to confirm.

**Comment by rajivml** [comment]
@jessesuen you are right, the mistake I am doing was, am editing an environment variable in a child process and exiting in the parent process based on the edit that I have made in the child process but the parent process won't be aware of this edit if we don't source the child shell script https://stackoverflow.com/questions/16217064/change-environment-variable-in-child-process-bash

You can close this issue and thanks for confirming that it should work
